### ISSUE TITLE
Dependency injection not working for SqlServerConnectionProvider

### ISSUE THREAD

**Issue by briankitt** [grandchamp/Identity.Dapper#79]
the project as I downloaded it, works fine. But I needed to customize it a bit.
I took the project here, and incorporated it into a new .net Core 2.1 project.  In the process, I somehow broke it.  The part I broke is related to dependency injection, and I'm not seeing the issue.

In my Startup.CS, 
`services.ConfigureDapperConnectionProvider<SqlServerConnectionProvider>(Configuration.GetSection("DapperIdentity"))
.ConfigureDapperIdentityCryptography(Configuration.GetSection("DapperIdentityCryptography"))
.ConfigureDapperIdentityOptions(new DapperIdentityOptions { UseTransactionalBehavior = false }); 
`
I'm debugging it, and when I get to the code below, the  _connectionProviderOptions is null.  On the sample project provided, it works fine.  As far as I can tell, both projects are identical, so I can't see what I broke.  The one difference is that the provided project SDK is NetStandard.Library (2.0.3), and my project SDK is Microsoft.NetCore.App (2.1).  So that does pull in different versions of the dependent libraries.  I suspect that is why it's broken.  Where I am confused, is that when I put the dependency in, I put it in as SqlServerConnectionProvider.  When it's refenced below, it's referenced as ConnectionProviderOptions.  Granted, I'm new to dependency injection, but to me, that should be null, because they are different types.  But in the app provided, it does somehow convert them.

`namespace Identity.Dapper.SqlServer.Connections
{
    public class SqlServerConnectionProvider : IConnectionProvider
    {
        private readonly IOptions<ConnectionProviderOptions> _connectionProviderOptions;
        private readonly EncryptionHelper _encryptionHelper;
        public SqlServerConnectionProvider(IOptions<ConnectionProviderOptions> connProvOpts, EncryptionHelper encHelper)
        {
            _connectionProviderOptions = connProvOpts;
            _encryptionHelper = encHelper;
        }

        public DbConnection Create()
        {
            if (_connectionProviderOptions.Value == null)
                throw new ArgumentNullException("There's no DapperIdentity configuration section registered. Please, register the section in appsettings.json or user secrets.");

`

**Comment by grandchamp** [comment]
Hi @briankitt. Can you provide a repro for your problem? Maybe you can upload it in your github.

**Comment by briankitt** [comment]
Ok.  I apologize.  This is the oddest thing.  I deleted my projects, and started over from scratch.  And it's now all working perfect.  I really don't know what I did to break it, but clearly I built the project wrong some how.   I worked for easily 3 hours trying to figure out what I did wrong, and just starting over fixed it.

Gremlins......

I would just delete this ticket, but I don't see a way to do that.

Thanks for your time, I appreciate your rapid feedback.
